File: server/src/main/java/gg/mineral/server/network/packet/play/clientbound/SignEditorOpenPacket.java
package gg.mineral.server.network.packet.play.clientbound;

import gg.mineral.api.network.packet.Packet;
import io.netty.buffer.ByteBuf;

public final record SignEditorOpenPacket(int x, int y, int z) implements Packet.OUTGOING {
    @Override
    public void serialize(ByteBuf os) {
        os.writeInt(x);
        os.writeInt(y);
        os.writeInt(z);
    }

    @Override
    public byte getId() {
        return 0x36;
    }
}

File: server/src/main/kotlin/gg/mineral/server/network/packet/play/clientbound/SignEditorOpenPacket.kt
package gg.mineral.server.network.packet.play.clientbound

import gg.mineral.api.network.packet.Packet
import io.netty.buffer.ByteBuf

@JvmRecord
data class SignEditorOpenPacket(val x: Int, val y: Int, val z: Int) : Packet.OUTGOING {
    override fun serialize(os: ByteBuf) =
        os.writeInt(x, y, z)

    override val id: Byte
        get() = 0x36
}
